server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/RootNode.java

```diff
@@ -15,24 +15,44 @@
 package org.eclipse.lsp.cobol.core.model.tree;
 
 import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Multimap;
 import lombok.Getter;
 import lombok.ToString;
 import org.eclipse.lsp.cobol.core.model.Locality;
 import org.eclipse.lsp.cobol.core.model.SyntaxError;
+import org.eclipse.lsp.cobol.core.semantics.NamedSubContext;
 import org.eclipse.lsp.cobol.service.utils.SyntaxTreeUtil;
+import org.eclipse.lsp4j.Location;
+import org.eclipse.lsp4j.Range;
 
-import java.util.List;
+import java.util.*;
 import java.util.stream.Collectors;
 
 import static org.eclipse.lsp.cobol.core.model.tree.NodeType.ROOT;
+import static org.eclipse.lsp.cobol.service.PredefinedCopybooks.IMPLICIT_PATH;
+import static org.eclipse.lsp.cobol.service.PredefinedCopybooks.PREF_IMPLICIT;
 
-/** The class represents the root. All trees must start with one root node. */
+/**
+ * The class represents the root. All trees must start with one root node. This class also creates
+ * definition nodes for copyNodes and adds its usages.
+ */
 @ToString(callSuper = true)
 @Getter
 public class RootNode extends Node {
-  public RootNode(Locality locality) {
+  private final Map<String, CopyDefinition> copyDefinitionMap = new HashMap<>();
+  public RootNode(Locality locality, NamedSubContext copybook) {
     super(locality, ROOT);
     addProcessStep(this::waitForVariableStructure);
+    createCopyBookDefinitionNode(copybook.getDefinitions());
+  }
+
+  private void createCopyBookDefinitionNode(Multimap<String, Location> definition) {
+    for (Map.Entry<String, Location> copybook : definition.entries()) {
+      copyDefinitionMap.put(
+              copybook.getKey(),
+          new CopyDefinition(
+              Locality.builder().range(copybook.getValue().getRange()).uri(copybook.getValue().getUri()).build(), copybook.getKey()));
```

Comment:
You don't need to convert `Location` to `Locality` here because it `CopyDefinition` you need to convert it back to `Location`. I think that you can save `Location` as is in the `CopyDefinition`